### PR TITLE
Json services

### PR DESCRIPTION
**Pull request by mostafa-abdullah**

### CODE REVIEW

**Comment by loayelzobeidy** [comment]
There is inconsistency between the data in the entities and  the responses in the company profile and the joblisting  the array should be of only one object to be consistent with the other entities. Also, why instead of having an api "localhost/api/unfriend" and another one   "localhost/api/friend" to have just one api "localhost/api/friend" with two different functions Post for adding a friend and DELETE for unfriend.

File: web_services/DELETE/delete_cv.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "request": {
+    "title": "DELETE CV",
+    "description": "remove CV from your profile",
+    "url": "http://localhost/api/cv",
+    "method": "DELETE",
+    "data": {
+      }
+  },
+  "response": {
+    "code": "Number"
```
Comment (Ahmed-anwar) [inline web_services/DELETE/delete_cv.json]: status_code

File: web_services/GET/companyProfile.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            "adminUserName":"String",
+            "adminUserID":"Number",
+            "adminUserIDMongo":"Number",
+            "industryType":"String",
+            "companyLocation":"String",
+            "relatedConnections":[{"userID":"Number","firstName":"String","lastName":"String","currentPostion":"String","profilePictureURL":"String"}],
+            "aboutUs":"String",
+            "website":"String/URL",
+            "yearFounded":"Date",
+            "headquarters":["String","string","string"],
```
Comment (Ahmed-anwar) [inline web_services/GET/companyProfile.json]: ["String", "String", "String"]

File: web_services/GET/jobListing.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+    "request": {
+        "title": "GET job listing",
+        "description": "return a job post information",
+        "url": "http://localhost/api/jobs/:jobID",
+        "method": "GET",
+    },
+    "response": {
+        "status_code": "number",
```
Comment (Ahmed-anwar) [inline web_services/GET/jobListing.json]: Number

File: web_services/GET/notifications.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+    "request": {
+        "title": "GET notifications",
+        "description": "get notifications of a specific user",
+        "url": "http://localhost/api/notifications/:userID",
+        "method": "GET",
+    },
+    "response": {
```
Comment (Ahmed-anwar) [inline web_services/GET/notifications.json]: Types should start with a capital letter for consistent convention.

File: web_services/PUT/edit_company_profile.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+          "headquarters":["String"],
+          "companytype":"String",
+          "companySize":"Integer",
+          "specialities":["String"],
+          "posts":[{"postID":"Number","authorID":"Number","media":{"images": ["String"],"videos": ["String"],"urls": ["String"]}}],
+          "jobListings":[{"jobID":"Number","positionName":"String","companyName":"String","companyID":"String","companyProfilePictureURL":"String"}]
+        }
+    },
+    "response": {
+        "status_code": "number"
```
Comment (Ahmed-anwar) [inline web_services/PUT/edit_company_profile.json]: Number

File: web_services/PUT/edit_job_as_company.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+          "compnayPicture":"URL/String",
+          "relatedConnections":[{"userID":"Number","firstName":"String","lastName":"String","currentPostion":"String","profilePictureURL":"String"}],
+          "jobBrief":"String",
+          "requiredSkills":["String"],
+          "note":"String",
+          "similarJobs":[{"jobID":"Number","positionName":"String","companyName":"String","companyID":"String","companyProfilePictureURL":"String"}]
+        }
+    },
+    "response": {
+        "status_code": "number"
```
Comment (Ahmed-anwar) [inline web_services/PUT/edit_job_as_company.json]: Number

File: web_services/DELETE/delete_like.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+  "request": {
+    "title": "delete like",
```
Comment (Ahmed-anwar) [inline web_services/DELETE/delete_like.json]: Delete

File: web_services/DELETE/delete_reply.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{
+  "request": {
+    "title": "delete reply",
```
Comment (Ahmed-anwar) [inline web_services/DELETE/delete_reply.json]: Delete

**Comment by HatemMorgan** [comment]
Great work :+1: 

In my opinion, I think we should add the issuer of the post requests as a query string.  

1- **accepted_friend_request**: does not have the issuer of the friend request the body only contains a user_id  which is properly the other user who send the request.
2-  **add_cv**, **add_friend** and **block_user** in Post folder have the same problem the I stated above.
3-  **unfriend** in Delete folder also has the same problem
4- I think we should change `"status_code": "Number"` to be: 
    -  `"status_code": "201"` for POST requests responses
    -  `"status_code": "200"` for GET requests responses
    -  `"status_code": "201"` for PUT requests responses
    -  `"status_code": "202"` for DELETE requests responses

**Comment by mostafa-abdullah** [comment]
@HatemMorgan we discussed the issuer thing. We can get his data via authentication, but we can leave it like this since we don't know yet how authentication will be handled.

For the status codes, I don't think it should be changed, because practically you don't know what status code you will get from the server (You might get a `503` due to a server error). So I think keeping it generic and consistent (by only specifying the type) is better.

**Comment by HatemMorgan** [comment]
@mostafa-abdullah getting issuer via authentication is a good idea :+1: 

I assumed that the responses written are for successful requests.

Lets merge :muscle:

**Pull request merged**